### DISCUSSION TITLE
How to read the file list of mini server by page

### DISCUSSION THREAD

**Discussion by InstantWindy** [minio/minio#12412] [Q&A]
It is too slow to read the files in the bucket at one time,how to read file list by page.

**Comment by dale0525** [comment]
Agreed. I have around 10K directories in a bucket and it takes over 10 seconds to list ALL of the items - sometimes only displays an error - while I just want to search for a single item.
I think we need something like Amazon S3's paging, only showing the first items, say like 20, and hide the remaining to other pages.
